scipy/sparse/data.py

```diff
@@ -167,31 +172,74 @@ def _min_or_max(self, axis, min_or_max):
 
         if axis < 0:
             axis += 2
+
         if (axis == 0) or (axis == 1):
             return self._min_or_max_axis(axis, min_or_max)
         else:
-            raise ValueError("invalid axis, use 0 for rows, or 1 for columns")
-
-    def max(self, axis=None):
-        """Maximum of the elements of this matrix.
+            raise ValueError("axis out of range")
 
+    def max(self, axis=None, out=None):
+        """
+        Return the maximum of the matrix or maximum along an axis.
         This takes all elements into account, not just the non-zero ones.
 
+        Parameters
+        ----------
+        axis : {-2, -1, 0, 1, None} optional
+            Axis along which the sum is computed. The default is to
+            compute the maximum over all the matrix elements, returning
+            a scalar (i.e. `axis` = `None`).
+
+        out : None, optional
+            This argument is in the signature *solely* for NumPy
+            compatibility reasons for the time being. Do not pass
```

Comment:
The phrases "for the time being" and "at the moment" are a bit speculative, and should be removed.